core/src/main/java/ai/timefold/solver/core/impl/heuristic/move/AbstractMove.java

```diff
@@ -7,26 +7,24 @@
 
 import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
 import ai.timefold.solver.core.api.score.director.ScoreDirector;
+import ai.timefold.solver.core.impl.move.director.VariableChangeRecordingScoreDirector;
 
 /**
  * Abstract superclass for {@link Move}, requiring implementation of undo moves.
- * Unless raw performance is a concern, consider using {@link AbstractSimplifiedMove} instead.
  *
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  * @see Move
  */
 public abstract class AbstractMove<Solution_> implements Move<Solution_> {
 
-    @Override
-    public final Move<Solution_> doMove(ScoreDirector<Solution_> scoreDirector) {
-        var undoMove = createUndoMove(scoreDirector);
-        doMoveOnly(scoreDirector);
-        return undoMove;
-    }
-
     @Override
     public final void doMoveOnly(ScoreDirector<Solution_> scoreDirector) {
-        doMoveOnGenuineVariables(scoreDirector);
+        // LegacyMoveAdapter does not wrap the score director in a VariableChangeRecordingScoreDirector
```

Comment:
I'm unsure if the comment makes sense. It looks like we will always use `VariableChangeRecordingScoreDirector`.